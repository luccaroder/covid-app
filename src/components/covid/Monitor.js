import moment from 'moment-timezone';
import React, { useState } from 'react';
import Button from '@material-ui/core/Button';
import httpClient from 'rapidApi'
import CommonDate from 'components/common/CommonDate'
import LineChart from 'components/covid/LineChart'
import { monitorStyles } from 'styles/monitor';
import { BuildTotalInformation } from 'builders/monitor';

export default function Monitor() {
    const styles = monitorStyles();
    const [selectedDate, setSelectedDate] = useState(new Date());
    const [chartInformation, setChartInformation] = useState([]);

    const handleDateChange = (date) => {
        setSelectedDate(date);
    };

    const handlechartInformation = (covidInformation) => {
        setChartInformation(covidInformation);
    };

    const getCovidCasesData = () => {
        const date = moment(selectedDate).format('YYYY-MM-DD');
        httpClient.get('/history_by_particular_country_by_date.php', { params: { country: 'Brazil', date }})
            .then(({data:{stat_by_country}}) => stat_by_country ?
                handlechartInformation(BuildTotalInformation(stat_by_country)) : [])
            .catch((error) => {
                console.log(error)
            })

    }

    return (
        <>
        <h1>Gráficos mostrando a evolução do Covid-19 no Brasil</h1>
        <h2>Informe um dia para ver os detalhes dos casos</h2>
        <div className={styles.root}>
            <CommonDate
                handleDateChange={handleDateChange}
                selectedDate={selectedDate}/>
            <Button variant="contained" color="primary" onClick={getCovidCasesData}>
                Pesquisar
            </Button>
            <LineChart chartInformation={chartInformation}/>
        </div>
        </>
    );
}