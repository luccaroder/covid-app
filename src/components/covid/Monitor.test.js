import React from 'react';
import Monitor from 'components/covid/Monitor'

afterEach(cleanup)

jest.mock('components/common/CommonDate', () => () => 'mock-common-date');
jest.mock('components/covid/LineChart', () => () => 'mock-line-chart');

test('renders Monitor react component', () => {
    const { asFragment } = render(<Monitor/>);
    expect(asFragment(<Monitor/>)).toMatchSnapshot();
});
