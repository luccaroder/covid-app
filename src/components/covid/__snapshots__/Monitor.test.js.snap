// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Monitor react component 1`] = `
<DocumentFragment>
  <h1>
    Gráficos mostrando a evolução do Covid-19 no Brasil
  </h1>
  <h2>
    Informe um dia para ver os detalhes dos casos
  </h2>
  <div
    class="makeStyles-root-1"
  >
    mock-common-date
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-label"
      >
        Pesquisar
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
    mock-line-chart
  </div>
</DocumentFragment>
`;
